File: Cpp/1656_Design_an_Ordered_Stream.cpp
class OrderedStream {
public:
    OrderedStream(int n) : streams(n + 1), ptr(1){
        
    }
    
    vector<string> insert(int id, string value) {
        streams[id] = value;
        vector<string> result;
        
        if(ptr != id) return result;
        
        for(; ptr < streams.size() && !streams[ptr].empty(); ++ptr) {
            result.push_back(streams[ptr]);
        }
        
        return result;
    }
private:
    vector<string> streams;
    int ptr;
};

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream* obj = new OrderedStream(n);
 * vector<string> param_1 = obj->insert(id,value);
 */

File: Kotlin/1656_Design_an_Ordered_Stream.kt
class OrderedStream(n: Int) {
    private var streams = Array<String>(n + 1) {""}
    private var ptr = 1
    
    fun insert(idKey: Int, value: String): List<String> {
        val result = mutableListOf<String>()
        streams[idKey] = value
        
        if (ptr != idKey) return result
        
        while (ptr < streams.size && streams[ptr].isNotEmpty()) {
            result.add(streams[ptr++])
        }
        
        return result
    }

}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * var obj = OrderedStream(n)
 * var param_1 = obj.insert(idKey,value)
 */